code/game/jobs/job/civilian.dm

```diff
@@ -284,6 +284,7 @@
 					H.equip_or_collect(new /obj/item/clothing/under/rank/clown(H), slot_w_uniform)
 					H.equip_or_collect(new /obj/item/clothing/shoes/clown_shoes(H), slot_shoes)
 		H.real_name = pick(clown_names)
+		H.dna.real_name = H.real_name
```

Comment:
The ordering there is weird: Clown spawn with normal char name, is given a random clown name for real_name, then gets the option to rename themself, which does fully_replace_character_name with the chosen name, UNLESS they pick an invalid name for their three attempts, in which case it just returns. I did the dna name set here to cover that situation.